server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -183,6 +353,165 @@ class DeltaSharedTableKernel(
+      }
+      val writerFeatures = if (p.getWriterFeatures.isEmpty) {
+        None
+      } else {
+        Some(p.getWriterFeatures.asScala.toSet)
+      }
+      DeltaFormatResponseProtocol(
+        deltaProtocol = DeltaProtocol(
+          minReaderVersion = p.getMinReaderVersion,
+          minWriterVersion = p.getMinWriterVersion,
+          readerFeatures = readerFeatures,
+          writerFeatures = writerFeatures
+        )
+      ).wrap
+    } else {
+      Protocol(p.getMinReaderVersion).wrap
+    }
+  }
+
+  private def getResponseMetadata(
+      m: KernelMetadata,
+      version: java.lang.Long,
+      respondedFormat: String): Object = {
+    // We override the metadata.id to be the table ID instead of the randomly generated ID defined
+    // by the Delta Lake protocol. The table ID is a UUID that uniquely identifies a shared table on
+    // the delta sharing server.
+    if (respondedFormat == DeltaSharedTableKernel.RESPONSE_FORMAT_DELTA) {
+      DeltaResponseMetadata(
+        deltaMetadata = DeltaMetadataCopy(
+          id = m.getId,
+          name = m.getName.orElse(null),
+          description = m.getDescription.orElse(null),
+          format = DeltaFormat(
+            provider = m.getFormat.getProvider,
+            options = m.getFormat.getOptions.asScala.toMap
+          ),
+          schemaString = cleanUpTableSchema(m.getSchemaString),
+          partitionColumns = getPartitionColumns(m),
+          configuration = getMetadataConfiguration(
+            respondedFormat = DeltaSharedTableKernel.RESPONSE_FORMAT_DELTA,
+            tableConf = m.getConfiguration.asScala.toMap),
+          createdTime = Option(m.getCreatedTime.orElse(null))
+        ),
+        version = version
+      ).wrap
+    } else {
+      Metadata(
+        id = m.getId,
+        name = m.getName.orElse(null),
+        description = m.getDescription.orElse(null),
+        format = Format(),
+        schemaString = cleanUpTableSchema(m.getSchemaString),
+        partitionColumns = getPartitionColumns(m),
+        configuration = getMetadataConfiguration(tableConf = m.getConfiguration.asScala.toMap),
+        version = version
+      ).wrap
+    }
+  }
+
+  // Get the partition columns from Kernel Metadata.
+  private def getPartitionColumns(metadata: KernelMetadata): Seq[String] = {
+    VectorUtils.toJavaList[String](metadata.getPartitionColumns).asScala
+  }
+
+  // Get the table configuration for parquet format response.
+  // If the response is in parquet format, return enableChangeDataFeed config only.
+  private def getMetadataConfiguration(
+      respondedFormat: String = DeltaSharedTableKernel.RESPONSE_FORMAT_PARQUET,
+      tableConf: Map[String, String]): Map[String, String ] = {
+    if (tableConfig.historyShared &&
+      tableConf.getOrElse("delta.enableChangeDataFeed", "false") == "true") {
+      if (respondedFormat ==  DeltaSharedTableKernel.RESPONSE_FORMAT_DELTA) {
+        Map("delta.enableChangeDataFeed" -> "true")
```

Comment:
@linzhou-db this line is a little bit hacky and I needed to make this change for this test case: getMetadata with configuration in io.delta.sharing.client.DeltaSharingRestClientDeltaSuite

Reply:
fixed